src/libraries/System.Text.Json/tests/Common/UnsupportedTypesTests.cs

```diff
@@ -0,0 +1,237 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public abstract class UnsupportedTypesTests : SerializerTests
+    {
+        private bool SupportsJsonPathOnSerialize { get; init; }
+        private bool SupportsIAsyncEnumerable { get; init; }
```

Comment:
Sure we can do that for "SupportsIAsyncEnumerable" but not "SupportsJsonPathOnSerialize" since JsonPath is not supported for fast-path serialize, and we test both fast-path and the serializer path here.